src/nplinker/nplinker.py

```diff
@@ -355,3 +356,55 @@ def save_data(
         data = (self.bgcs, self.gcfs, self.spectra, self.mfs, self.strains, links)
         with open(file, "wb") as f:
             pickle.dump(data, f)
+
+    def objects_to_tsv(self, objects: Sequence[BGC] | Sequence[Spectrum], filename: str) -> None:
+        """Exports a list of BGC or Spectrum objects to a specified file in tab-separated format.
+
+        Args:
+            objects (list): A list of BGC or a list of Spectrum objects to be exported.
+            filename (str): The name of the file where the data will be saved.
+        """
+        if not objects:
+            raise ValueError("No objects provided to export")
+
+        # Ensure all elements in the list are of the same type
+        obj_type = type(objects[0])
+        if not all(isinstance(obj, obj_type) for obj in objects):
+            raise TypeError("All objects in the list must be of the same type")
+
+        headers = objects[0].to_dict().keys()
+        with open(self._output_dir / filename, "w", newline="") as outfile:
+            writer = csv.DictWriter(outfile, fieldnames=headers, delimiter="\t")
+            writer.writeheader()
+            for obj in objects:
+                row = obj.to_dict()
+                for header in headers:
+                    value = row[header]
+                    # Convert list, tuple, set to comma-separated string
+                    if isinstance(value, (list, tuple, set)):
+                        row[header] = ", ".join(map(str, value))
+                    # Convert dict to comma-separated string
+                    elif isinstance(value, dict):
+                        row[header] = ", ".join([f"{k}:{v}" for k, v in value.items()])
+                    # Convert anything else to string
+                    else:
+                        row[header] = str(value) if value else ""
+                    # Replace tabs with 4 spaces
+                    row[header] = row[header].replace("\t", "    ")
```

Comment:
This part could be extracted to a `to_tabular` method with some modifications for both BGC and Spectrum classes.

Then you could use pandas dataframe to faciliate the process of a list of objects.

Reply:
I moved this part into a t`to_string` method. Let me know if that's okay as well.